src/tracks/Pattern.cpp

```diff
@@ -218,6 +219,7 @@ Note * Pattern::addNote( const Note & _new_note, const bool _quant_pos )
 	m_notes.insert(std::upper_bound(m_notes.begin(), m_notes.end(), new_note, Note::lessThan), new_note);
 	instrumentTrack()->unlock();
 
+	adjustSteps();
```

Comment:
Could you explain why you update `m_steps` only in this case, not whenever the length changes?

Reply:
When a user clicks on "Add steps" and creates a pattern on the added steps, if he then deselects all of the steps in the extended range for a moment, the range is shrunk immediately. I thought this would be a bit annoying, though it looks nicer when steps are moved inside the piano roll or removed and the beat pattern adapts. This would be a one line change essentialy.

Reply:
I could be wrong, that's my first glance on the PR. But from what I read, on master `m_steps` is only changed by methods related to the BBTrack. When you have an `MelodyPattern` (adding notes through the piano roll), it seems that the length of the pattern changes but `m_steps` remains the same value. So when you delete the notes and it goes back to being a `BeatPattern`, the number of steps is the one you set earlier on the BBEditor. I didn't test yet but I think with this line you'd end up updating the number of steps as you add notes and once you go back to a `BeatPattern` you'd have the last value you had, not the one you set on the BBEditor. Changing that behavior is not really necessary and IMHO it makes sense to keep `BeatPattern` number of steps separated from the `MelodyPattern` length.